apps/bridge-dapp/src/hooks/useAddCurrency.ts

```diff
@@ -23,12 +24,70 @@ const getCurrencyImageUrl = async (symbol: string): Promise<string> => {
   }
 };
 
+export const addTokenAddedToMetamaskInLocalStorage = (
+  accountAddress: string,
+  resourceId: string,
+  tokenAddress: string
+) => {
+  // Retrieve existing token storage
+  const storedTokens = JSON.parse(localStorage.getItem('addedTokens') || '{}');
+
+  // Check if account address exists in storage
+  if (!storedTokens[accountAddress]) {
+    storedTokens[accountAddress] = {};
+  }
+
+  // Check if resourceId exists for the account
+  if (!storedTokens[accountAddress][resourceId]) {
+    storedTokens[accountAddress][resourceId] = {};
+  }
+
+  // Check if the token is already marked as added for the account and resourceId
+  if (!storedTokens[accountAddress][resourceId][tokenAddress]) {
+    storedTokens[accountAddress][resourceId][tokenAddress] = true;
+    localStorage.setItem('addedTokens', JSON.stringify(storedTokens));
+  }
+};
+
+export const isTokenAddedToMetamask = (
+  currency: Currency,
+  activeChain?: Chain,
+  activeApi?: any,
+  accountAddress?: string
+): boolean => {
+  // Validate required parameters.
+  if (!currency || !activeChain || !activeApi || !accountAddress) {
+    return false;
+  }
+
+  const typedChainId = calculateTypedChainId(
+    activeChain.chainType,
+    activeChain.chainId
+  );
+
+  const tokenAddress = currency.getAddressOfChain(typedChainId);
+
+  // Get resourceId from config if it exists, otherwise return false.
+  const resourceId =
+    activeApi?.config?.currencies[currency.view.id]?.addresses?.get(
+      typedChainId
+    );
```

Comment:
I think there is a misunderstanding here. The `addresses` in the `currencies[currencyId]` is the anchor addresses record, not the resource id. You should use the resource id from the `useCurrentResourceId` (in `libs/react-hooks/src/useCurrentResourceId.ts`).